lib/publiccloud/azure.pm

```diff
@@ -143,4 +143,32 @@ sub ipa {
     return $self->run_ipa(%args);
 }
 
+sub on_terraform_timeout {
+    my ($self) = @_;
+    my $out = script_output('terraform state show azurerm_resource_group.openqa-group');
+    if ($out !~ /name\s+=\s+(openqa-[a-z0-9]+)/m) {
+        record_info('ERROR', 'Unable to get resource-group:' . $/ . $out, result => 'fail');
+        return;
+    }
+    my $resgroup = $1;
+
+    my $tries = 3;
+    while ($tries gt 0) {
+        $tries = $tries - 1;
+        eval {
+            my $bootlog_name = '/tmp/azure-bootlog.txt';
+            $out = script_output('az vm boot-diagnostics enable --ids $(az vm list -g ' . $resgroup . ' --query \'[].id\' -o tsv) --storage ' . $self->storage_account, 60 * 5, proceed_on_failure => 1);
+            record_info('INFO', $out);
+            script_run('az vm boot-diagnostics get-boot-log --ids $(az vm list -g ' . $resgroup . ' --query \'[].id\' -o tsv) > ' . $bootlog_name);
+            upload_logs($bootlog_name, failok => 1);
+            $tries = 0;
+        };
+        if ($@) {
+            type_string(qq(\c\\));
+        }
+    }
+
+    assert_script_run("az group delete --yes --no-wait --name $resgroup");
```

Comment:
should we also add a timeout in this command? Sometimes Azure is also slow deleting things...

Reply:
I covered it by adding `--no-wait`, so this command actually returns immediately.